### ISSUE TITLE
Don't say "No update available." when updater_server fails

### ISSUE THREAD

**Issue by taam** [nextcloud/updater#403]
Currently the updater_server service always(?) fails with HTTP code 500, see: https://github.com/nextcloud/updater_server/issues/698

However, the updater seems to interpret this as "no update available" and suggests a manual update, which is misleading. (I've seen some forum entries where this has caused problems, it cost me quite some time, and it might even lead to people damaging their instances trying to upgrade manually).

I'm not sure if this affects Nextcloud (server) as well, it somehow found out there's an update and showed the correct version. (Does is have a different way to search for updates?)

**Comment by joshtrichards** [comment]
> However, the updater seems to interpret this as "no update available" and suggests a manual update, which is misleading. (

Agreed. While we monitor the connection success, you're right that we aren't checking for a 500 error in the response. So our current implementation doesn't see a problem as long as the HTTPS server accepts the connection (particularly since the response is permitted to be empty).

We need code similar to this from `downloadUpdate()`:

https://github.com/nextcloud/updater/blob/4af30bdab591f6d7e9ad51b36d1b6525e5c1bf86/lib/Updater.php#L569-L597

Added within `getUpdateServerResponse()` just after this line:

https://github.com/nextcloud/updater/blob/4af30bdab591f6d7e9ad51b36d1b6525e5c1bf86/lib/Updater.php#L510

> I'm not sure if this affects Nextcloud (server) as well, it somehow found out there's an update and showed the correct version. (Does is have a different way to search for updates?)

Server, at least when using the standard `updatenotification` app, has a background job that checks for updates once a day. And, yes, it checks for updates through an entirely independent code path since the Updater app is completely independent from Server.
